plugins/reports/counter/classes/LegacyJR1.inc.php

```diff
@@ -0,0 +1,358 @@
+
+		return array($begin, $end);
+	}
+
+	/**
+	 * Get the valid journal IDs for which log entries exist in the DB.
+	 * @param $useLegacyStats boolean Use the old counter plugin data.
+	 * @return array
+	 */
+	function _getJournalIds($useLegacyStats = false) {
+		$metricsDao =& DAORegistry::getDAO('MetricsDAO'); /* @var $metricsDao MetricsDAO */
+		if ($useLegacyStats) {
+			$results = $metricsDao->getMetrics(OJS_METRIC_TYPE_LEGACY_COUNTER, array(STATISTICS_DIMENSION_ASSOC_ID));
+			$fieldId = STATISTICS_DIMENSION_ASSOC_ID;
+		} else {
+			$filter = array(STATISTICS_DIMENSION_ASSOC_TYPE => ASSOC_TYPE_GALLEY);
+			$results = $metricsDao->getMetrics(OJS_METRIC_TYPE_COUNTER, array(STATISTICS_DIMENSION_CONTEXT_ID), $filter);
+			$fieldId = STATISTICS_DIMENSION_CONTEXT_ID;
+		}
+		$journalIds = array();
+		foreach($results as $record) {
+			$journalIds[] = $record[$fieldId];
+		}
+		return $journalIds;
+	}
+
+	/**
+	 * Retrieve a monthly log entry range.
+	 * @param $journalId int
+	 * @param $begin
+	 * @param $end
+	 * @param $useLegacyStats boolean Use the old counter plugin data.
+	 * @return 2D array
+	 */
+	function _getMonthlyLogRange($journalId = null, $begin, $end, $useLegacyStats = false) {
+		$begin = date('Ym', strtotime($begin));
+		$end = date('Ym', strtotime($end));
+
+		$metricsDao =& DAORegistry::getDAO('MetricsDAO'); /* @var $metricsDao MetricsDAO */
+		$columns = array(STATISTICS_DIMENSION_MONTH, STATISTICS_DIMENSION_FILE_TYPE);
+		$filter = array(
+			STATISTICS_DIMENSION_MONTH => array('from' => $begin, 'to' => $end)
+		);
+
+		if ($useLegacyStats) {
+			$dimension = STATISTICS_DIMENSION_ASSOC_ID;
+			$metricType = OJS_METRIC_TYPE_LEGACY_COUNTER;
+		} else {
+			$dimension = STATISTICS_DIMENSION_CONTEXT_ID;
+			$metricType = OJS_METRIC_TYPE_COUNTER;
+			$filter[STATISTICS_DIMENSION_ASSOC_TYPE] = ASSOC_TYPE_GALLEY;
+		}
+
+		if ($journalId) {
+			$columns[] = $dimension;
+			$filter[$dimension] = $journalId;
+		}
+
+		$results = $metricsDao->getMetrics($metricType, $columns, $filter);
+		return $results;
+	}
+
+	/**
+	 * Retrieve a monthly log entry range.
+	 * @param $begin
+	 * @param $end
+	 * @param $useLegacyStats boolean Use the old counter plugin data.
+	 * @return 2D array
+	 */
+	function _getMonthlyTotalRange($begin, $end, $useLegacyStats = false) {
+		return $this->_getMonthlyLogRange(null, $begin, $end, $useLegacyStats);
+	}
+
+	/**
+	 *
+	 * Counter report in XML
+	 * @param $request PKPRequest
+	 * @param $year string
+	 * @param $useLegacyStats boolean
+	 */
+	function _reportXML(&$request, $year, $useLegacyStats) {
+		$templateManager =& TemplateManager::getManager();
+		list($begin, $end) = $this->_getLimitDates($year);
+
+		$this->_assignTemplateCounterXML($templateManager, $begin, $end, $useLegacyStats);
+		$templateManager->display($this->getParentPlugin()->getTemplatePath() . 'reportxml.tpl', 'text/xml');
+	}
+
+	/**
+	 * Get the parent plugin
+	 * @return object
+	 */
+	function &getParentPlugin() {
```

Comment:
Reading it again, I think it's an unnecessary coupling you have between this class and the plugin one. The plugin class knowing the interface of this class is ok, but this class doesn't need to know the plugin interface. It can receive just the template path as a parameter in the constructor. Also, why not implementing this as a CounterReport?